### DISCUSSION TITLE
Reflective boundary conditions

### DISCUSSION THREAD

**Discussion by ghislainp** [lululxvi/deepxde#1348] [Q&A]
Hi,

I've a first order ODE
D [y1, y2] / Dt = f(y1,y2)

and I'd like to constrain y1 = 0.5 * y2 at one boundary. Ideally with a hard constraint. Do you have any suggestion ?

**Comment by lululxvi** [comment]
Seems not easy to do.

**Comment by ghislainp** [comment]
In fact, it seems to work, with a soft constraint:

```python
def boundary_condition_bottom_func(x, y, X):
    return y[:, 0] - 0.5 * y[:, 1]

bc_bottom = dde.OperatorBC(geom, boundary_condition_bottom_func, boundary_condition_bottom)
```

Do you think it is fine ?

**Reply by lululxvi** [reply]
It is fine.
